src/transformers/tools/agents.py

```diff
@@ -219,6 +220,11 @@ def __init__(self, chat_prompt_template=None, run_prompt_template=None, addition
 
         self.prepare_for_new_chat()
 
+    @property
+    def toolbox(self) -> Dict[str, Tool]:
+        """Get all tool currently available to the agent"""
+        return self._toolbox
+
```

Comment:
What is the benefit of this design?

Reply:
To avoid the user changing `agent.toolbox`. They can still change a key, but not replace (and accidentally eras) the whole variable.